packages/client/src/pages/Game.tsx

```diff
@@ -1,20 +1,90 @@
-import { useEffect, useState } from 'react';
+/** @jsx jsx */
+/** @jsxRuntime classic */
+
+import { PropsWithChildren, MouseEvent, useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { Box, Flex, Text } from 'theme-ui';
-import { Suit } from 'shared';
+import { Box, Flex, Text, jsx } from 'theme-ui';
 import { Card } from 'components/Cards/Card';
 import { Deck } from 'components/Cards/Deck';
 import { Link } from 'react-router-dom';
 import { Layout } from 'components/Layout';
 import { gameIO } from 'lib/socket';
 import { RoomState, RoomEvent } from 'shared';
-import { Card as CardType } from 'utils/cardEngine';
+import { Card as CardType, deck } from 'utils/cardEngine';
 import { useUserData } from 'components/UserContext';
 import { GameTable } from 'components/GameTable';
+import { shuffle, sum } from 'lodash';
+import { Suit } from 'shared';
+
+type CardWrapperProps = PropsWithChildren<{ onClick: (e: MouseEvent<HTMLElement>) => void; isActive: boolean }>;
+const CardTableWrapper = ({ isActive, ...props }: CardWrapperProps) => (
+  <Box
+    role="button"
+    sx={{
+      position: 'relative',
+      '&::after': isActive
+        ? {
+            content: "''",
+            position: 'absolute',
+            top: 0,
+            bottom: 0,
+            left: 0,
+            right: 0,
+            border: '2px solid',
+            borderColor: 'primary',
+            borderRadius: 1,
+          }
+        : undefined,
+    }}
+    {...props}
+  />
+);
+
+const CardWrapper = (props: CardWrapperProps) => (
+  <CardTableWrapper
+    sx={{
+      transition: 'transform 0.3s ease-in-out',
+      transform: props.isActive ? 'translateY(-3.5vw)' : undefined,
+      '&:hover': {
+        transform: 'translateY(-3.5vw)',
+      },
+    }}
+    {...props}
+  />
+);
+
+const useCardState = () => {
+  const shuffledDeck = shuffle(deck());
+  const opponent = shuffledDeck.splice(0, 3);
+  const player = shuffledDeck.splice(0, 3);
+  const table = shuffledDeck.splice(0, 4);
+  const [cardState, setCardState] = useState({
+    shuffledDeck,
+    opponent,
+    player,
+    table,
+    opponentStack: [] as CardType[],
+    playerStack: [] as CardType[],
+  });
+  return {
+    ...cardState,
+    setCardState,
+  };
+};
+
+const cardKey = (card: CardType) => `${card[0]}__${card[1]}`;
```

Comment:
Wouldn't it be better if Cart was an object? and then it would be `card.number` (if that's allowed) and `card.suit`.

Reply:
it's a tuple, we can make it an object yes but it should work fine like that